### PR TITLE
update connectedmachine

### PR DESCRIPTION
**Pull request by vedkale**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by vedkale** [comment]
I had a question about updating the "tags" property for all our resources - for some reason this property is called "tag" and not "tags" and I can't figure out how to update it. Please let me know how I could fix it

File: src/ConnectedMachine/ConnectedMachine.Autorest/custom/New-AzConnectedMachineRunCommand_ScriptLocalPath.ps1 [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+                    }
+                    $script += ";"
+                }
+            }
+            
+            $PSBoundParameters.Add("SourceScript", $script)
+            # If necessary, remove the -ParameterA parameter from the dictionary of bound parameters
+            $null = $PSBoundParameters.Remove("ScriptLocalPath")
+        }
+        Az.ConnectedMachine\New-AzConnectedMachineRunCommand @PSBoundParameters
```
Comment (wyunchi-ms) [inline src/ConnectedMachine/ConnectedMachine.Autorest/custom/New-AzConnectedMachineRunCommand_ScriptLocalPath.ps1:298]:
```suggestion
            $Tag = $PSBoundParameters['Tag']
            $null= $PSBoundParameters.Remove("Tag")
            $null = $PSBoundParameters.Add("Tags", $Tag)

        Az.ConnectedMachine\New-AzConnectedMachineRunCommand @PSBoundParameters
```

**Comment by vedkale** [comment]
@wyunchi-ms could you help me address this [issue](https://github.com/Azure/azure-powershell/issues/19664)

**Comment by wyunchi-ms** [comment]
> @wyunchi-ms could you help me address this [issue](https://github.com/Azure/azure-powershell/issues/19664)

This issue is about the response's property. Not the input parameter. We accept the `tags` in response model. Since the response  model is generated from swagger. Usually it's because swagger defined the property as `tags`. If you want to modify it, it's better to confirm with the service team member.

**Comment by vedkale** [comment]
I am a part of the service team, just trying to figure out how I would make the change.

**Comment by wyunchi-ms** [comment]
Do you mean change the swagger or service response?

**Comment by vedkale** [comment]
We use the common types from swagger, so the service sends tags but for some reason powershell module shows tag instead

**Comment by wyunchi-ms** [comment]
Could you please show me the link of swagger?

**Comment by vedkale** [comment]
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-10-03-preview/HybridCompute.json

**Comment by wyunchi-ms** [comment]
Do you mean: https://github.com/Azure/azure-rest-api-specs/blob/68030f1c17e0cd5912e1bda17a5f28473e84b5f9/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2023-10-03-preview/HybridCompute.json#L3549 it's `tags`

**Comment by vedkale** [comment]
That is used for resource updates and [this](https://github.com/Azure/azure-rest-api-specs/blob/68030f1c17e0cd5912e1bda17a5f28473e84b5f9/specification/common-types/resource-management/v3/types.json#L61C8-L61C18) is used to represent the resource. But for some reason the powershell response shows `tag` instead of `tags`

image

**Comment by wyunchi-ms** [comment]
Please follow the step: https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#property-rename

File: src/ConnectedMachine/ConnectedMachine.Autorest/README.md [hunk 2]
```diff
@@ -265,6 +265,12 @@ directive:
       parameter-name: AgentUpgradeEnableAutomaticUpgrade
     set:
       parameter-name: AgentUpgradeEnableAutomatic
+    
+  # Rename Tag to Tags
```
Comment (msJinLei) [inline src/ConnectedMachine/ConnectedMachine.Autorest/README.md:269]: We prefer to use a singular name for parameter. Further, I don't the doc update for `Tag`

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by msJinLei** [comment]
Please also fix the failed test case.

**Comment by msJinLei** [comment]
@vedkale LGTM, please leave a message to describe your change here.

**Review by msJinLei — APPROVED** [review]

**Comment by vedkale** [comment]
- Added -ScriptLocalPath to New-AzConnectedMachineRunCommand to let users add script files locally.
- Added machineName parameter to the McahineExtension and MachineRunCommand models

**Pull request merged**